tests/test_main.py

```
@@ -607,6 +607,221 @@ def _capture_create_task(coro: Any, *args: Any, **kwargs: Any) -> Any:
+        """
+        Early sync failures should be handled before startup drain completes.
+        """
+        shutdown_event = _OnePassEvent()
+        startup_drain_complete_event = threading.Event()
+        readiness_calls: list[str] = []
+
+        mock_matrix_client = AsyncMock()
+        mock_matrix_client.add_event_callback = MagicMock()
+        mock_matrix_client.close = AsyncMock()
+
+        async def _sync_forever_fail_once(*_args: Any, **_kwargs: Any) -> None:
```

Ruff (0.15.9)

[warning] Dynamically typed expressions (typing.Any) are disallowed in `*_args`

(ANN401)